aten/src/ATen/native/native_functions.yaml

```diff
@@ -1733,6 +1733,9 @@
   variants: method
   device_guard: false
 
+- func: one_hot(IndexTensor self, int64_t num_classes=0) -> Tensor
```

Comment:
Let's change the default to `num_classes=-1` and throw on all other non-negative values. `-1` in PyTorch in many cases (view, etc) can mean "infer this number" so I think it is a better value to use than 0